src/Serialization/serializers.jl

```diff
@@ -112,7 +112,8 @@ function save_data_basic(s::SerializerState, x::Any,
   !isnothing(key) && set_key(s, key)
   begin_node(s)
   str = string(x)
-  write(s.io, "\"$str\"")
+  #write(s.io, "\"$str\"")
+  JSON.print(s.io, str)
```

Comment:
```suggestion
  JSON.show_string(s.io, str)
  nothing
```
This would avoid a few allocations and should be faster.

Reply:
Why add nothing on 117?

Reply:
Because `show_string` always returns 1, the return value of `write(io, "\"")`. And we don't want a return value? (Maybe it won't hurt but I think it is better to discard it directly here instead of assuming that this happens further down in the call stack)

Reply:
```
julia> str = "sjdlkdsfjslkdjflew kewj dsf sdf\n\nsd lkdsjfldsfsdf\"sdjlkfdsf\ndsfdsf";

julia> @benchmark JSON.show_string(devnull, str)
BenchmarkTools.Trial: 10000 samples with 989 evaluations.
 Range (min … max):  43.068 ns … 74.669 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     51.306 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   51.370 ns ±  1.652 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

                                   ▁▄▇█▇▇▆▆▇▆▄▃▂▁              
  ▂▁▁▁▁▁▂▁▂▂▂▂▂▂▂▂▂▂▂▃▂▂▂▃▃▃▃▃▃▃▄▅▆███████████████▇▇▆▅▅▄▄▃▃▂▃ ▄
  43.1 ns         Histogram: frequency by time        55.2 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.

julia> @benchmark JSON.print(devnull, str)
BenchmarkTools.Trial: 10000 samples with 741 evaluations.
 Range (min … max):  168.926 ns …   3.081 μs  ┊ GC (min … max): 0.00% … 92.51%
 Time  (median):     189.308 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   207.036 ns ± 203.506 ns  ┊ GC (mean ± σ):  8.08% ±  7.59%

                ▁▃▇█▅▅▅▃▂                                        
  ▁▁▁▁▁▁▂▁▁▄▇▅▅▆██████████▅▄▃▃▂▂▂▂▂▂▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▃
  169 ns           Histogram: frequency by time          236 ns <

 Memory estimate: 432 bytes, allocs estimate: 5.
```